tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -50,18 +51,17 @@ using tensorflow::tracing::ScopedAnnotation;
 // Implementation note: HLO profiling is always enabled for GPU executables,
 // since we can use timers around thunks.
 GpuExecutable::GpuExecutable(
-    const string& ptx, const std::vector<uint8>& cubin,
-    std::pair<int, int> compute_capability,
-    std::unique_ptr<const ThunkSchedule> thunk_schedule,
+    const string& text, const std::vector<uint8>& binary,
+    GpuVersion gpu_version, std::unique_ptr<const ThunkSchedule> thunk_schedule,
```

Comment:
Can we use two factory functions `GpuExecutable::CreateForNvidia` and `GpuExecutable::CreateForAmd`, and let them take different version parameters?

Reply:
@timshen91 This would create additional bifurcations later on so I'm not a fan of this idea.

In another PR https://github.com/tensorflow/tensorflow/pull/30660 I'm attempting to unify logic between `nvptx_compiler.cc` and its AMDGPU-equivalent by forming a base class `GpuCompiler` which extracts common HLO lowering logic. Having two separate factory functions would mean the construction of `GpuExecutable` instances would then fall on the concrete subclasses of `GpuCompiler`, `NVPTXCompiler` / `AMDGPUCompiler` (and `MlirCompiler`).

Reply:
@timshen91 Also please refer to #30326 to see how `GpuVersion` is used to drive different LLVM backends.

Reply:
Agreed. Then I think it should be a complete custom type (e.g. a class) instead of a std::variant. The underlying data member can be a variant, though.

Reply:
Based on actual use cases in #30660 and #30326 `GpuVersion` is really an `std::variant/absl::variant` with no extra additional use cases. Would it be fine to let be a type alias for now, and evolve to a full custom type later on when new use cases arise?

Reply:
Sure.

Reply:
Looks like we are resolved on the data members in `GpuExecutable` for now. Marking this as resolved.